### ISSUE TITLE
Future Elasticsearch Image plans

### ISSUE THREAD

**Issue by mleklund** [upmc-enterprises/elasticsearch-operator#276]
Wondered if there was a plan to handle the fact that the pires/docker-elasticsearch-kubernetes repo is abandoware at this point.

**Comment by while1eq1** [comment]
I built my own here that I shared that supports ES 6.3+

https://github.com/while1eq1/elasticsearch-kubernetes-searchguard

**Comment by pieterlange** [comment]
What makes you say pires/docker-elasticsearch-kubernetes is deprecated? Seems supported to me.

**Comment by kskewes** [comment]
I forked @while1eq1 repo above for my use and haven't made any major changes yet. Thanks very much for your work there.

I think we should adopt it for this repo as it works well and reduces the number of FROM statements.

**Comment by while1eq1** [comment]
Also the pires one wouldnt work with this operator if the ES version was 6.3+ as the SearchGuard SSL standalone plugin was rolled into mainline SearchGuard and doesnt support anything after ES 6.2.

This required changes such as initializing searchguard at runtime and adding all the sgconfigs.

Also, to use my ES repo with this you need to build the elasticsearch-operator from master branch

**Comment by mleklund** [comment]
Also, with the open sourcing of xpack, should consideration be given to basing the image on the upstream projects docker image?  Docker hub's "official" image is based on elastic.co's image now.  There are size considerations of an alpine based image vs another platform, but it could pay off in stability as elastic's image likely gets a large amount of testing.

**Comment by pieterlange** [comment]
xpack isn't "true" open source though (elastic license), as we've discussed in another issue here or in pires' repo so i'm not sure about packaging that.

I tried building my own ES 6.1.3 image with a recent searchguard a couple of months ago but also ran into some upgrading issues due to the searchguard support model changing. Ran out of my timebox and didn't have time yet to look back.
This is a really useful project and i've used it a lot personally, it would make sense for us to all ask for some company time to contribute to maintaining this project. Getting us over this searchguard/xpack hurdle being one of the first bigger things to cover.

If anyone is up for spending some time on this, specifically on maintaining https://github.com/pires/kubernetes-elasticsearch-cluster/ and related docker image, ping me on slack @gianrubio @while1eq1 @komljen @mumoshu @skokovic

**Comment by stevesloka** [comment]
I think we should rally around @while1eq1's image. Do you want to make that a thing @while1eq1? 

I also might see about moving this to a generic org/repo. Would be nice to have all the code in one place (e.g. operator, cron, image, etc)

**Comment by while1eq1** [comment]
@stevesloka Yes, Id be 100% down for that, lets make it happen.

**Comment by skokovic** [comment]
@pieterlange I'm up for it, don't know what slack you're referring to though.

**Comment by dimm0** [comment]
Is it gonna happen?
(he's referring to https://slack.com messenger)

**Comment by dimm0** [comment]
Ah.. I see :) no idea.

**Comment by mleklund** [comment]
https://www.elastic.co/blog/security-for-elasticsearch-is-now-free

**Comment by stevesloka** [comment]
Yeah I saw that @mleklund, the question now is should this operator exist now that elastic has an official one?

**Comment by while1eq1** [comment]
Once the official one is tested and known to provide all the needed
features id imagine the answer would be no. What is your thoughts?

**Comment by stevesloka** [comment]
That's my thinking, but need to investigate the current solution to verify features. I don't think it makes sense to invest in multiple solutions that accomplish the same thing and given there's a vendor solution, I think it makes sense to move to that.

**Comment by komljen** [comment]
Zalando team have their own ES operator, but it seems the main focus is on scaling https://github.com/zalando-incubator/es-operator. Of course, it would be great if we have only one operator to rule them all, but I don't see that happen anytime soon.

**Comment by komljen** [comment]
> I also might see about moving this to a generic org/repo. Would be nice to have all the code in one place (e.g. operator, cron, image, etc)

@stevesloka any updates on this?

**Comment by kfox1111** [comment]
> https://www.elastic.co/blog/security-for-elasticsearch-is-now-free

Free but not open. Its in the basic tier, not the open source tier.

The open distro folks seem to be maintaining images too. maybe those would work?
https://hub.docker.com/r/amazon/opendistro-for-elasticsearch

**Comment by dimm0** [comment]
I tried installing the official one, never worked for me... Still using yours

**Comment by komljen** [comment]
> The open distro folks seem to be maintaining images too. maybe those would work?

It works with the operator with minor modifications, https://github.com/komljen/opendistro-for-elasticsearch-k8s

I removed the security plugin here.
